File: android/ssabree/app/src/main/java/com/ssafy/ssabree/core/datasource/remote/model/PostRequests.kt
package com.ssafy.ssabree.core.datasource.remote.model

data class PostCreateRequest(
    val title: String,
    val content: String,
    val boardId: Long,
    val imageUrls: List<String>,
    val poll: PollCreateRequest?
)

data class PostUpdateRequest(
    val title: String?,
    val content: String?
)

data class CommentCreateRequest(
    val content: String
)

data class ReplyCreateRequest(
    val content: String
)

data class VoteRequest(
    val optionId: Long
)

data class PollCreateRequest(
    val title: String,
    val options: List<String>
)

File: iOS/ssabree/ssabree/Core/Network/DTO/PostRequests.swift
import Foundation

struct PostCreateRequest: Codable {
    let title: String
    let content: String
    let boardId: Int
    let imageUrls: [String]
    let poll: PollCreateRequest?
}

struct PollCreateRequest: Codable {
    let title: String
    let options: [String]
}

struct PostUpdateRequest: Codable {
    let title: String
    let content: String
}

struct CommentCreateRequest: Codable {
    let content: String
}

struct ReplyCreateRequest: Codable {
    let content: String
}

struct VoteRequest: Codable {
    let optionId: Int
}
